inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -1131,6 +1131,55 @@ void FuseMultipleIdentitiesPass::run() {
     }
 }
 
+void BroadcastConstPass::run() {
+    for (auto& constLayer : *pLayers) {
+        if (!LayerInfo(constLayer).isConst()) {
+            continue;
+        }
+        auto isNonFunctional = [](CNNLayerPtr l) {
+            return LayerInfo(l).isNonFunctional();
+        };
+        if (!CNNNetHasNextLayerSkipCertain(constLayer, 0, 0, isNonFunctional)) {
+            continue;
+        }
+
+        auto nextLayer = CNNNetGetNextLayerSkipCertain(constLayer, 0, 0, isNonFunctional).first;
+
+        if (!LayerInfo(nextLayer).isEltwise()) {
+            continue;
+        }
+
+        auto constDims = constLayer->outData.front()->getTensorDesc().getDims();
+        auto constDimsSize = InferenceEngine::details::product(constDims.begin(), constDims.end());
+        auto eltwiseDims = nextLayer->outData.front()->getTensorDesc().getDims();
+        auto eltwiseDimsSize = InferenceEngine::details::product(eltwiseDims.begin(), eltwiseDims.end());
+
+        if (constDimsSize == eltwiseDimsSize) {
+            continue;
+        }
+        if (eltwiseDimsSize % constDimsSize != 0) {
+            continue;
+        }
+
+        auto currentConstBlob = constLayer->blobs.begin()->second;
```

Comment:
blobs might be empty or missed "custom" key that requires for const blobs - can we check that us well here